Modules/Chatroom/classes/class.ilChatroomFormFactory.php

```diff
@@ -82,96 +88,135 @@ private function addDefaultBehaviour(ilPropertyFormGUI $form): ilPropertyFormGUI
-        $title = new ilTextInputGUI($this->lng->txt('title'), 'title');
-        $title->setRequired(true);
-        $form->addItem($title);
-
-        $description = new ilTextAreaInputGUI($this->lng->txt('description'), 'desc');
-        $form->addItem($description);
-
-        $section = new ilFormSectionHeaderGUI();
-        $section->setTitle($this->lng->txt('rep_activation_availability'));
-        $form->addItem($section);
-
-        $online = new ilCheckboxInputGUI($this->lng->txt('rep_activation_online'), 'online_status');
-        $online->setInfo($this->lng->txt('chtr_activation_online_info'));
-        $form->addItem($online);
-
-        $dur = new ilDateDurationInputGUI($this->lng->txt('rep_time_period'), 'access_period');
-        $dur->setShowTime(true);
-        $form->addItem($dur);
-
-        $visible = new ilCheckboxInputGUI($this->lng->txt('rep_activation_limited_visibility'), 'access_visibility');
-        $visible->setValue('1');
-        $visible->setInfo($this->lng->txt('chtr_activation_limited_visibility_info'));
-        $dur->addSubItem($visible);
-
-        $presentationHeader = new ilFormSectionHeaderGUI();
-        $presentationHeader->setTitle($this->lng->txt('settings_presentation_header'));
-        $form->addItem($presentationHeader);
-
-        $objectService->commonSettings()->legacyForm(
-            $form,
-            $chatroom
-        )->addTileImage();
-
-        $num_msg_history = new ilNumberInputGUI($this->lng->txt('display_past_msgs'), 'display_past_msgs');
-        $num_msg_history->setSuffix($this->lng->txt('display_past_msgs_suffix'));
-        $num_msg_history->allowDecimals(false);
-        $num_msg_history->setSize(5);
-        $num_msg_history->setInfo($this->lng->txt('hint_display_past_msgs'));
-        $num_msg_history->setMinValue(0);
-        $num_msg_history->setMaxValue(100);
-        $form->addItem($num_msg_history);
-
-        $cb_history = new ilCheckboxInputGUI($this->lng->txt('chat_enable_history'), 'enable_history');
-        $cb_history->setInfo($this->lng->txt('chat_enable_history_info'));
-        $form->addItem($cb_history);
-
-        $functionsnHeader = new ilFormSectionHeaderGUI();
-        $functionsnHeader->setTitle($this->lng->txt('chat_settings_functions_header'));
-        $form->addItem($functionsnHeader);
-
-        $cb = new ilCheckboxInputGUI($this->lng->txt('allow_anonymous'), 'allow_anonymous');
-        $cb->setInfo($this->lng->txt('anonymous_hint'));
-        $form->addItem($cb);
-
-        $cb = new ilCheckboxInputGUI($this->lng->txt('allow_custom_usernames'), 'allow_custom_usernames');
-        $txt = new ilTextInputGUI($this->lng->txt('autogen_usernames'), 'autogen_usernames');
-        $txt->setRequired(true);
-        $txt->setInfo($this->lng->txt('autogen_usernames_info'));
-        $cb->addSubItem($txt);
-        $form->addItem($cb);
-
-        return $form;
+        return $this->refinery->custom()->transformation(static function (array $values): array {
```

Comment:
Can be simplified to:
```suggestion
        return $this->refinery->custom()->transformation(static fn (array $values): array => array_merge(...$values));
```